### PR TITLE
WIP: docs: add documentation for cue and rego

### LINKED ISSUE #868: Documentation and Examples for cue/rego policies!
Let's add some documentation and examples for attestations in general, but in particular with cue and rego policies.

For example, we should clarify exactly what gets passed into the validator (just the predicate portion?), and show a few getting started examples.

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-by: Furkan Türkal <furkan.turkal@trendyol.com>

#### Summary

This PR will add documentation for cue and rego

#### Ticket Link

Fixes #868 

#### Release Note

```release-note
Docs: add documentation for cue and rego
```

### CODE REVIEW

**Comment by dlorenc** [comment]
> @dlorenc what do you think about the @lukehinds comment above, do we move this documentation to the `sigstore` website, if so, I can close this one and start another PR for the `sigstore `website.

I think we need to do a big bulk move at some point, moving just this over won't really help. It's fine to get this in somewhere so it's at least documented as we figure out how to get these rendered on the main website.

Do we have an issue for migrating everything?

**Comment by lukehinds** [comment]
please port changes here (depending on which lands first) https://github.com/sigstore/sigstore-website/pull/67

File: EXAMPLES.md [hunk 1]
```diff
@@ -45,4 +46,113 @@ The following checks were performed on each of these signatures:
...
+ before:    "2021-10-09T17:10:27Z"
+ Data:      "bar"
+ Timestamp: >"2021-10-09T17:10:27Z"
+}
+There are 1 number of errors occurred during the validation:
+- Data: conflicting values "foo\n" and "bar"
+Error: 1 validation errors occurred
+```
+
+### [Cosign Custom Predicate](./specs/COSIGN_PREDICATE_SPEC.md) type and Rego policy
```
Comment (developer-guy) [inline EXAMPLES.md:105]: @srenatus Is there anything you would like to add?

File: README.md [hunk 2]
```diff
@@ -311,6 +311,31 @@ To verify:
...
 
+##### Validate In-Toto Attestations
+
+`cosign` has support of validating In-toto Attestations against `CUE` and `Rego` policies.
+
+Let's start with giving a brief introduction to `CUE` and `Rego` who might not be familiar with these.
+
+* [CUE](https://cuelang.org) is an open-source data validation language and inference engine with its roots in logic programming.
+
+* [Rego](https://www.openpolicyagent.org/docs/latest/policy-language/) was inspired by Datalog, which is a well understood, decades old query language. Rego extends Datalog to support structured document models such as JSON.
```
Comment (srenatus) [inline README.md:322]: 🤔 I'd cut this down a bit:
```suggestion
* [Rego](https://www.openpolicyagent.org/docs/latest/policy-language/) was inspired by Datalog, and supports structured document models such as JSON.
```

Really wouldn't want to give the impression that CUE's logic foundation was _not well understood_ or anything. Less is more 😄

**Pull request closed without merging**